logic/tool/TypeResolver.java

```diff
@@ -170,10 +178,52 @@ private TypeVariable register(TypeVariable var) {
             }
             resolverRegister.put(var.id(), resolver);
             variableRegister.putIfAbsent(resolver.reference(), var);
+            if (!var.resolvedTypes().isEmpty()) traversal.labels(resolver.id(), var.resolvedTypes());
```

Comment:
`ResolveLabels` already dealt with `LabelConstraints`, so we just need to pass over the results.